### DISCUSSION TITLE
Migration Issue

### DISCUSSION THREAD

**Discussion by bamidele2161** [prisma/prisma#23151] [Q&A]
### Question

Error: migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20240212172140_main` migration started at 2024-02-15 09:48:23.665218 UTC failed.

The solution here, https://pris.ly/d/migrate-resolve says I should run prisma migrate, which will cause data loss

any other solution please?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by bamidele2161** [comment]
@ludralph 
any help please?

**Reply by AshutoshSajan** [reply]
I guess you need to rollback the migration to timestamp_main

**Reply by nurul3101** [reply]
@bamidele2161 Did you try to follow this section of the docs to rollback you migration? It doesn't mention invoking `migrate dev` command:
https://www.prisma.io/docs/orm/prisma-migrate/workflows/patching-and-hotfixing#moving-backwards-and-reverting-all-changes

Also, what exactly is causing the migrations to fail?

**Reply by gchen1196** [reply]
if you have access to your production database, delete the pending migration row in `_prisma_migration` table. and then rerun prisma migrate deploy

**Comment by bamidele2161** [comment]
@AshutoshSajan 
Thanks for the response.
To rollback, I need to run npx prisma migrate dev which will lead to data loss
